example-doc-generator/python/publish_sample.py

```
@@ -0,0 +1,542 @@
+def commit_and_push(
+    samples_repo: Path,
+    project_name: str,
+    branch_name: str,
+    dry_run: bool,
+) -> None:
+    if dry_run:
+        dry(f"git add . && git commit -m 'samples: add {project_name} connector integration sample'")
+        dry(f"git push origin {branch_name}")
+        return
+    info("Committing changes...")
+    subprocess.run(["git", "add", "."], cwd=str(samples_repo), check=True)
```

Ruff (0.15.7)

[error] Starting a process with a partial executable path

(S607)